src/Database.js

```diff
@@ -165,6 +123,148 @@
+							min = Math.min(min,self.data[j][d]);
+							max = Math.max(max,self.data[j][d]);
+						}
+					}
+					self.dimensionDomains[d] = [min,max];
+				}
+			}
+			//Anything else is a string type
+			else {
+				self.dimensionTypes[d] = CINEMA_COMPONENTS.DIMENSION_TYPE.STRING;
+				self.dimensionDomains[d] = self.data.map(function(p){return p[d];});
+			}
+		});
+	};
+
+	/**
+	 * Function that refreshes the data.  If there are changes, the dataUpdated dispatcher is called.
+	 * @param {bool} reloadAllData - Whether or not the refreshData should load the full file (true) data or
+	 *                               just check for file size changes (false)
+	 */
+	CINEMA_COMPONENTS.Database.prototype.refreshData = function(reloadAllData) {
+		var self = this;
+
+		if (reloadAllData) {
+			// Check all data in the file
+			getAndParseCSV(self.path, function(data_arr, request) { self.dataUpdateCallback(data_arr, request); }, self.errorCallback);
+		}
+		else {
+			// Only check for file size changes
+			var xhReq = new XMLHttpRequest();
+			xhReq.open("HEAD", self.path, true);
+			xhReq.onreadystatechange = function() {
+				if (xhReq.readyState === 4) {
+					if (xhReq.status === 200 || 
+						//Safari returns 0 on success (while other browsers use 0 for an error)
+						(navigator.userAgent.match(/Safari/) && xhReq.status === 0)
+					) {
+										
+						var contentLength = xhReq.getResponseHeader('Content-Length');
+										
+						if (contentLength != self.prevContentLength) {
+							getAndParseCSV(self.path, function(data_arr, request) { self.dataUpdateCallback(data_arr, request); }, self.errorCallback);
+						}
+					}
+				}
+			}
+		
+			xhReq.send(null);
+		}
+	}
+
+	/**
+	 * Callback Function that checks if data has changed after loading the file.
+	 * @param {string} data_arr = The data from the file.
+	 * @param {object} request = The request where we can get the response header information
+	 */
+	CINEMA_COMPONENTS.Database.prototype.dataUpdateCallback = function(data_arr, request) {
+		var self = this;
+
+		// Get new content length
+		self.prevContentLength = request.getResponseHeader('Content-Length');	
+
+		//Convert rows from arrays to objects
+		var newData = data_arr.slice(1).map(function(d) {
+			var obj = {};
+			self.dimensions.forEach(function(p,i){obj[p] = d[i];});
+			return obj;
+		});
+
+		// Determine whether there has been a change in the data
+		var updated = false;
+		var updateInfo = { added: [], modified: [], removed: [], oldData: self.data, oldDimensionDomains: self.dimensionDomains };
+		for (var f = 0; f < self.data.length || f < newData.length; f++) {
+			if (f >= self.data.length) {
+				updateInfo.added.push(f);
+				updated = true;
+			}
+			else if (f >= newData.length) {
+				updateInfo.removed.push(f);
+				updated = true;
+			}
+			else if (!(JSON.stringify(self.data[f]) === JSON.stringify(newData[f])) ) {
+				updateInfo.modified.push(f);
+				updated = true;
+			}
+		}
```

Comment:
I'm not sure the updateInfo object gives very useful information about what has actually changed. Consider what would happen if the very first row of data is removed. updateInfo would report every row as having been changed and then that the last row has been removed, which doesn't really accurately reflect the actual change. Also, we already changed the database's data and most components use d3's data binding system which already has the ability to act on added, changed and removed data individually, so I don't think this updateInfo object is necessary.

Reply:
Okay, I'll look into it.  For some reason Bandit needed to know if new rows were added and knowing which ones these were, is definitely helpful because I only wanted to load new rows, not the entire database.

Reply:
Ah okay, that probably has to do with how Bandit loads information iteratively (at least that's what it did last time I worked on it). I'll leave it up to your discretion to find something that plays nice with Bandit.

Reply:
I kept the object that same for now, but I'm open to suggestions.